internal/updater/checker.go

```diff
@@ -119,22 +125,37 @@ func (u *Checker) GetUpdateInfo(desiredChannel, desiredVersion string) (*Availab
 		if code == 404 || strings.Contains(string(res), "Could not retrieve update info") {
 			// The above string match can be removed once https://www.pivotaltracker.com/story/show/179426519 is resolved
 			logging.Debug("Update info 404s: %v", errs.JoinMessage(err))
+			u.an.EventWithLabel(anaConst.CatConfig, anaConst.ActUpdateCheck, anaLabelUnavailable, &dimensions.Values{
+				Version: p.StrP(desiredVersion),
+				Error:   p.StrP("Update info 404"),
+			})
 			return nil, nil
 		} else if code == 403 || code == 503 {
 			// The request could not be satisfied or service is unavailable. This happens when Cloudflare
 			// blocks access, or the service is unavailable in a particular geographic location.
 			logging.Warning("Update info request blocked or service unavailable: %v", err)
+			u.an.EventWithLabel(anaConst.CatConfig, anaConst.ActUpdateCheck, anaLabelUnavailable, &dimensions.Values{
+				Version: p.StrP(desiredVersion),
+				Error:   p.StrP("Update info request blocked or service unavailable"),
+			})
 			return nil, nil
 		}
+		u.an.EventWithLabel(anaConst.CatConfig, anaConst.ActUpdateCheck, anaLabelFailed, &dimensions.Values{
+			Version: p.StrP(desiredVersion),
+			Error:   p.StrP("Could not fetch update info"),
+		})
 		return nil, errs.Wrap(err, "Could not fetch update info from %s", infoURL)
 	}
 
-	info := &AvailableUpdate{}
+	info := &AvailableUpdate{
+		an: u.an,
```

Comment:
I agree it's awkward but I'm not sure any of the alternatives are better. We could pass the analytics instance to the `AvailableUpdate`'s methods or create a new constructor. Either way, since this object is responsible for downloading and installing the update I think we want the analytics in there.